utils/paramfetch/src/lib.rs

```diff
@@ -48,18 +48,28 @@ struct ParameterData {
 }
 
 #[inline]
-fn param_dir() -> String {
-    std::env::var(DIR_ENV).unwrap_or_else(|_| PARAM_DIR.to_owned())
+fn param_dir(mut data_dir: PathBuf) -> PathBuf {
+    if let Ok(path_buf_string) = std::env::var(PathBuf::from(DIR_ENV)) {
+        PathBuf::from(path_buf_string)
+    } else {
+        data_dir.push(PARAM_DIR);
+        data_dir
+    }
```

Comment:
Perhaps something like:

```suggestion
    std::env::var(PathBuf::from(DIR_ENV))
        .map(PathBuf::from)
        .unwrap_or_else(|_| data_dir.join(PARAM_DIR))
```